suite/tests/api/ir_aarch64.c

```diff
@@ -458,6 +458,189 @@ test_fmov_general(void *dc)
     test_instr_encoding(dc, OP_fmov, instr);
 }
 
+static void
+test_fmov_vector(void *dc)
+{
+    byte *pc;
+    instr_t *instr;
+
+    /* FMOV <Vd>.<T>, #<imm> (v8.2 half-precision)
+     * 16 bit floating-point values encoded in instruction's 8 bit field, so
+     * there is a fixed, limited set of floating-point values which can be set,
+     * see 'Table C2-2 Floating-point constant values' in section 'Modified
+     * immediate constants in A64 floating-point instructions.' of the Arm
+     * Reference Manual.
+     */
+    instr = INSTR_CREATE_fmov_vector_imm(
+        dc, opnd_create_reg(DR_REG_Q1), opnd_create_immed_float(1.0), OPND_CREATE_HALF());
```

Comment:
nit suggest: Use `1.0f` for a closer literal type to what is present, so the user doesn't expect double precision, and we don't have unnecessary casting?  Ditto below for the non-doubles (and see above for adding double support).

Reply:
Done.

Reply:
run arm tests